### PR TITLE
Poll for latest XBTUSD BitMEX quote

### PR DESCRIPTION
**Pull request by luckysori**
Instead of using a WebSocket which is a connection that needs to be managed.

Yet to do:

- Rename `bitmex_price_feed` to `bitmex_quote`.
- Separate `wire` type `Quote` from `bitmex_quote::Quote`.

### CODE REVIEW

File: daemon/src/bitmex_price_feed.rs [hunk 1]
```diff
@@ -36,184 +65,172 @@ impl Actor {
...
-        pub bid_size: u64,
-        pub ask_size: u64,
-        pub bid_price: f64,
-        pub ask_price: f64,
-        pub symbol: String,
-        pub timestamp: String,
+    async fn get_latest_quote(&self) -> Result<Quote> {
+        let mut url = reqwest::Url::parse(BITMEX_QUOTE_URL).expect("valid URL");
+        let query = serde_urlencoded::to_string(GetQuoteBucketedRequest::latest_xbt_usd())
+            .expect("valid query");
```
Comment (thomaseizinger) [inline daemon/src/bitmex_price_feed.rs:169]: I'd probably just hardcode this URL with is query parameters, same as we did before!

Comment (thomaseizinger) [reply daemon/src/bitmex_price_feed.rs:169]: You should then be able to just use `reqwest::get`.

File: daemon/src/bitmex_price_feed.rs [hunk 2]
```diff
@@ -36,184 +65,172 @@ impl Actor {
...
+                    bail!("Failed to fetch latest quote from BitMEX: {name}, {message}")
+                }
+                Err(deser_err) => {
+                    tracing::warn!("Failed to deserialize BitMEX error: {}", deser_err);
+                    bail!("Failed to fetch latest quote from BitMEX: {}", content)
+                }
+            }
+        }
+
+        let quotes = match serde_json::from_str::<Vec<Quote>>(&content) {
```
Comment (thomaseizinger) [inline daemon/src/bitmex_price_feed.rs:196]: Could deserialize into a 1-element array.

File: daemon/src/bitmex_price_feed.rs [hunk 3]
```diff
@@ -36,184 +65,172 @@ impl Actor {
...
+        if !status.is_success() {
+            match serde_json::from_str::<BitmexErrorResponse>(&content) {
+                Ok(BitmexErrorResponse {
+                    error: BitmexErrorMessage { message, name },
+                }) => {
+                    bail!("Failed to fetch latest quote from BitMEX: {name}, {message}")
+                }
+                Err(deser_err) => {
+                    tracing::warn!("Failed to deserialize BitMEX error: {}", deser_err);
+                    bail!("Failed to fetch latest quote from BitMEX: {}", content)
```
Comment (thomaseizinger) [inline daemon/src/bitmex_price_feed.rs:191]: I wouldn't bother doing `resp.text()` just for the log message.

**Review by thomaseizinger — COMMENTED** [review]
Some ideas!

**Comment by luckysori** [comment]
Closed because [we think we have figured out](https://github.com/itchysats/itchysats/pull/1342) how to reliably restart the WebSocket-based price feed actor, so we don't need to rewrite it using HTTP requests.

**Pull request closed without merging**